### PR TITLE
CSS linting integrated into pre-commit

### PR DESCRIPTION
**Pull request by ryanahamilton**
- Adds [stylelint](https://stylelint.io/) CSS linter, utilizing the [`stylelint-config-standard` configuration](https://github.com/stylelint/stylelint-config-standard). This is a widely used base configuration—we are able to override the rules as we see fit in `.stylelintrc`.
- Integrates the linter into pre-commit hooks.
- Fixes (or ignores) all remaining violations yielded by the linter. `bootstrap-theme.css` was beyond full repair, so it has a couple blanket ignores. Given its vendor-like nature, that seemed like an appropriate course of action.

CC: @mik-laj as promised!

---
Make sure to mark the boxes below before creating PR: [x]

- [x] Description above provides context of the change
- [x] Unit tests coverage for changes (not needed for documentation changes)
- [x] Commits follow "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)"
- [x] Relevant documentation is updated including usage instructions.
- [x] I will engage committers as explained in [Contribution Workflow Example](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#contribution-workflow-example).

---
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/master/UPDATING.md).
Read the [Pull Request Guidelines](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#pull-request-guidelines) for more information.

### CODE REVIEW

**Comment by ashb** [comment]
@ryanahamilton Looking at the yarn lockfile do we now have babel 6 and babel 7? Another future PR to tidy that up perhaps?

**Review by kaxil — APPROVED** [review]

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -360,3 +360,9 @@ repos:
         entry: "./scripts/ci/pre_commit_bat_tests.sh"
         files: ^breeze$|^breeze-complete$|\.sh$|\.bash$
         pass_filenames: false
+      - id: stylelint
+        name: stylelint
+        entry: "stylelint"
+        language: node
+        files: ^airflow/www/.*\.(css|scss|sass)$
+        args: []
```
Comment (kaxil) [inline .pre-commit-config.yaml]: I think it might be safe to remove this `args` if it is just empty

Comment (ryanahamilton) [reply .pre-commit-config.yaml]: Ah, you're right. I had previously removed it but still had `pass_filenames` in there and it wasn't working. Just removed it.

**Comment by ryanahamilton** [comment]
@kaxil it appears that [Travis failed](https://travis-ci.org/github/apache/airflow/jobs/672618449) on the `stylelint` pre-commit hook with a:
```
Executable `stylelint` not found`
```
Is this a chicken-before-the-egg situation where it hasn't been installed to be executed yet?

**Comment by ashb** [comment]
> @kaxil it appears that [Travis failed](https://travis-ci.org/github/apache/airflow/jobs/672618449) on the `stylelint` pre-commit hook with a:
> 
> ```
> Executable `stylelint` not found`
> ```
> 
> Is this a chicken-before-the-egg situation where it hasn't been installed to be executed yet?

Oh, yes. We should find a minimal docker image that has stylelint installed in it. That's how we've handled this for other checks.

File: .pre-commit-config.yaml [hunk 2]
```diff
@@ -360,3 +360,8 @@ repos:
         entry: "./scripts/ci/pre_commit_bat_tests.sh"
         files: ^breeze$|^breeze-complete$|\.sh$|\.bash$
         pass_filenames: false
+      - id: stylelint
+        name: stylelint
+        entry: "stylelint"
+        language: node
+        files: ^airflow/www/.*\.(css|scss|sass)$
```
Comment (mik-laj) [inline .pre-commit-config.yaml:355]:
```suggestion
        files: ^airflow/www/.*\.(css|scss|sass)$
        additional_dependencies: ["stylelint@13.3.0"]
```
This should also work. Please look at the example:  https://github.com/pre-commit/mirrors-eslint/blob/master/.pre-commit-hooks.yaml

You can also use: https://github.com/awebdeveloper/pre-commit-stylelint

I don't know which solution is better, but downloading a large image with the system for one simple application can be expensive. I suspect that there is no one common base image for all pre-commit hooks.

File: airflow/www/package.json [hunk 3]
```diff
@@ -51,6 +51,8 @@
     "mini-css-extract-plugin": "^0.4.1",
     "moment-locales-webpack-plugin": "^1.2.0",
     "style-loader": "^0.21.0",
+    "stylelint": "^13.3.0",
```
Comment (mik-laj) [inline airflow/www/package.json]: Are these required dependencies of our application or dependencies of the development environment?

File: .pre-commit-config.yaml (hunk 2 above)
Comment (mik-laj) [reply .pre-commit-config.yaml:355]: Should we also add `stylelint-config-standard": "^20.0.0`?

Comment (ryanahamilton) [reply .pre-commit-config.yaml:355]: ah, yes, that needs to be added too. Thanks!

File: airflow/www/package.json (hunk 3 above)
Comment (ryanahamilton) [reply airflow/www/package.json]: They are development environment dependencies (`devDependencies`).

Comment (mik-laj) [reply airflow/www/package.json]: But you install it in a virtual environment provided by pre-commit. Do you also need it here the second time?

Comment (ryanahamilton) [reply airflow/www/package.json]: Are developers running pre-commit in their development environment (I'm not. Is that a thing?)? (I'm not well versed in pre-commit.)

I suppose it isn't technically required—though as a developer, I want to be checking against these requirements before making any local commits, and utilize the package in CLI and in conjunction with my IDE plugin.

Comment (mik-laj) [reply airflow/www/package.json]: Pre-commit downloads and installs the node  Most developers use pre-commit because it is checked in Travis. Without checking changes by pre-commit it cannot be accepted.
https://github.com/pre-commit/pre-commit/blob/master/pre_commit/languages/node.py#L59

I am only afraid that if you have the same app in two places, they may behave differently due to additional dependencies. The pre-commit should also integrate with the IDE because the IDE uses git, but I don't use git in the IDE.

Comment (mik-laj) [reply airflow/www/package.json]: Generally, this frontend is not intensively developed and it is possible that we need more advanced solutions, e.g. installing the node in Breeze and then starting the stylelint by breeze. 

https://github.com/apache/airflow/blob/master/.pre-commit-config.yaml#L195-L200
https://github.com/apache/airflow/blob/master/.pre-commit-config.yaml#L223-L229
https://github.com/apache/airflow/blob/master/.pre-commit-config.yaml#L336-L362

However, I have no opinion on this matter. If you prefer to have packages in package.json, this will ok for me, but it's worth adding a comment that we should keep two places in sync.

Comment (ashb) [reply airflow/www/package.json]: I would prefer to have them in package.json as well please.

Comment (ryanahamilton) [reply airflow/www/package.json]: Thank you for the additional context @mik-laj. I've left them in both, but added a comment as you suggested.

**Comment by potiuk** [comment]
Love it :)

File: airflow/providers/amazon/aws/secrets/secrets_manager.py [hunk 4]
```diff
@@ -0,0 +1,125 @@
+#
```
Comment (potiuk) [inline airflow/providers/amazon/aws/secrets/secrets_manager.py]: I think it needs rebase - those are unrelated changes.

Comment (kaxil) [reply airflow/providers/amazon/aws/secrets/secrets_manager.py]: Done :)

**Review by ashb — APPROVED** [review]
